packages/amplify-graphql-schema-generator/src/schema-generator/generate-schema.ts

```diff
@@ -0,0 +1,207 @@
+import { Field, Index, Model, Schema } from '../schema-representation';
+import { Kind, print } from 'graphql';
+import { FieldWrapper, ObjectDefinitionWrapper, DirectiveWrapper } from '@aws-amplify/graphql-transformer-core';
+
+export const generateGraphQLSchema = (schema: Schema): string => {
+  const models = schema.getModels();
+  const document: any = {
+    kind: Kind.DOCUMENT,
+    definitions: [],
+  };
+
+  models.forEach(model => {
+    const type = constructObjectType(model);
+    
+    const fields = model.getFields();
+    fields.forEach(f => {
+      const field: any = convertInternalFieldTypeToGraphQL(f);
+      type.fields.push(field);
+    });
+    
+    addPrimaryKey(type, model.getPrimaryKey());
+    addIndexes(type, model.getIndexes());
+
+    document.definitions.push(type.serialize());
+  });
+
+  const schemaStr = print(document);
+  return schemaStr;
+};
+
+const convertInternalFieldTypeToGraphQL = (field: Field): FieldWrapper => {
+  const typeWrappers = [];
+  let fieldType = field.type;
+  while (fieldType.kind !== "Scalar" && fieldType.kind !== "Custom") {
+    typeWrappers.push(fieldType.kind);
+    fieldType = (fieldType as any).type;
+  }
+
+  // construct the default directive
+  const fieldDirectives = [];
+  if (field.default) {
+    fieldDirectives.push(new DirectiveWrapper({
+      kind: Kind.DIRECTIVE,
+      name: {
+        kind: "Name",
+        value: "default",
+      },
+      arguments: [
+        {
+          kind: "Argument",
+          name: {
+            kind: "Name",
+            value: "value",
+          },
+          value: {
+            kind: "StringValue",
```

Comment:
Just a comment for future, is it possible to avoid constructing these GraphQL objects via big JSON inputs and do it via some API?